coderd/userauth.go

```diff
@@ -257,6 +322,9 @@ func (api *API) userOIDC(rw http.ResponseWriter, r *http.Request) {
 			Message: "Failed to extract OIDC claims.",
 			Detail:  err.Error(),
 		})
+		// We pass a disposable user ID just to force an audit diff
+		// and generate a log for a failed login
+		aReq.New = database.APIKey{UserID: uuid.New()}
```

Comment:
I think this might work if we just do this at the top:

```golang
aReq.Old = database.APIKey{}
defer func() {
	if aReq.New.UserID == uuid.Nil {
		aReq.New = database.APIKey{UserID: uuid.New()}
	}
}()
defer commitAudit()

```

Reply:
Good idea! Fixed!